posts/2020-09-29-microprofile-openapi-beta-200011.adoc

```diff
@@ -0,0 +1,253 @@
+MicroProfile OpenAPI 2.0 introduces some new annotations that simplify the process of generating OpenAPI documentation for your REST APIs. It also introduces a new MicroProfile Config property prefix that can be used to define the schema for Java classes that you do not have the source code for.
+
+[#schemaproperty]
+==== @SchemaProperty
+
+The @SchemaProperty annotation has been added to allow you to include properties in the definition of a schema when using the @Schema annotation. Previously, the @Schema annotation did not allow you to define any properties. Properties were introspected from the implementation class.
+With the introduction of the @SchemaProperty annotation, a list of properties can now defined using the properties attribute on the @Schema annotation, as follows:
+
+[source, java]
+----
+@Schema(
+    name = "User",
+        description = "Telephone number to contact the user”,
+        implementation = User.class,
+        properties = {
+            @SchemaProperty(
+                name = "phone",
+                description = "Telephone number to contact the user”
+            )
+        }
+    )} 
+----
+[#requestbodyschema]
+==== @RequestBodySchema
+
+The @RequestBodySchema annotation has been added to provide a shorthand mechanism to specify the schema for a request body. Previously, defining the schema for a request body required the use of three different annotations, as follows:
+[source, java]
+----
+@RequestBody(
+    content = {
+        @Content(
+            schema = @Schema(
+                implementation = MyRequestObject.class
+            )
+        )
+    }
+)
+----
+
+With the introduction of the @RequestBodySchema annotation, this can be simplified to the following:
+
+[source, java]
+----
+@RequestBodySchema(MyRequestObject.class) 
+----
+
+[#apiresponseschema]
+==== @APIResponseSchema
+
+The @APIResponseSchema annotation has been added to provide a shorthand mechanism to specify the schema for a response body. Previously, defining the schema for a response body required the use of three different annotations, as follows:
+
+[source, java]
+----
+@APIResponse(
+    content = {
+            @Content(
+                schema = @Schema(
+                    implementation = MyResponseObject.class
+                )
+            )
+        }
+    )
+----
+
+With the introduction of the @RequestBodySchema annotation, this can be simplified to the following:
```

Comment:
"to the following" -> "to the following single annotation"

Reply:
Done